File: backend/app/schemas/company.py
from pydantic import BaseModel, ConfigDict, Field
from uuid import UUID
from datetime import datetime
from typing import Optional

class CompanyBase(BaseModel):
    name: str = Field(..., description="The name of the company")
    domain: Optional[str] = None
    industry: Optional[str] = None
    company_size: Optional[str] = None
    source_added_from: Optional[str] = None

class CompanyCreate(CompanyBase):
    pass

class CompanyRead(CompanyBase):
    id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/company.ts
export interface Company {
  id: string;
  name: string;
  domain: string | null;
  industry: string | null;
  company_size: string | null;
  source_added_from: string | null;
  created_at: string;
  updated_at: string;
}

export interface CompanyCreate {
  name: string;
  domain?: string;
  industry?: string;
  company_size?: string;
  source_added_from?: string;
}
